kasa/discover.py

```diff
@@ -273,9 +274,27 @@ async def discover_single(
         """
         loop = asyncio.get_event_loop()
         event = asyncio.Event()
+
+        try:
+            ipaddress.ip_address(host)
+            ip = host
+        except ValueError:
+            try:
+                adrrinfo = await loop.getaddrinfo(
+                    host,
+                    0,
+                    type=socket.SocketKind.SOCK_DGRAM,
+                    family=socket.AddressFamily.AF_INET,
+                )
+                ip = adrrinfo[0][4][0]
```

Comment:
Is there a better way to access the info? Having a three-times nested array accesses feel error-prone to me..

Reply:
It's actually quite stable due to the data getaddrinfo returns.  I've added the following comment before it to make this clear:

```
getaddrinfo returns a list of 5 tuples with the following structure:
(family, type, proto, canonname, sockaddr)
where sockaddr is 2 tuple (ip, port).
hence [0][4][0] is a stable array access because if no socket
address matches the host for SOCK_DGRAM AF_INET the gaierror
would be raised.
https://docs.python.org/3/library/socket.html#socket.getaddrinfo
```

Reply:
Ah, indeed. My bad, that's all good :+1: